micromanager_gui/main_window.py

```diff
@@ -220,16 +224,11 @@ def update_max_min(self, event=None):
 
         self.tab_wdg.max_min_val_label.setText(min_max_txt)
 
-    def snap(self):
+    def _snap(self, img: np.ndarray):
+        # _snap in a thread so we don't freeze UI
+        create_worker(self.update_viewer, img, _start_thread=True)
```

Comment:
As it stands this won't actually snap a new image - it just updates the viewer from a thread. I think what you want here is to create a thread worker to call snap just like in the button widget (sorry if I was confusing about that earlier)

Reply:
True! But since the image is snapped through the `pymmcore-plus` `snap` method in `snap_button.py` and is "emitted" with the `pymmcore-plus` `imageSnapped` signal, here in main_window we just need to display it, correct?

Reply:
So `_snap` is used for the `snap_on_click` from the stage widget. So this needs to call snap for those. 

So I think there needs to be a  `_snap` calls `snap` from a thread, and then you can directly hook up `update_viewer` to the `imageSnapped` signal - that doesn't need to have a thread be created.

Reply:
> So `_snap` is used for the `snap_on_click` from the stage widget. So this needs to call snap for those.

as I mention in #142, I love this "snap_after_stage_moves" function but I think we can remove it for now and see how to add it back once the stage, the snap and the live button widgets are merged. Do you agree?